packages/snaps-execution-environments/src/common/utils.ts

```diff
@@ -57,3 +59,40 @@ export async function withTeardown<T>(
       });
   });
 }
+
+/**
+ * Creates and returns specific type of Proxy used for limiting StreamProvider.
+ *
+ * @param provider - Instance of a StreamProvider to be limited.
+ * @param request - Custom attenuated request object.
+ * @returns Proxy to the StreamProvider instance.
+ */
+export function proxyStreamProvider(
+  provider: StreamProvider,
+  request: unknown,
+): StreamProvider {
+  // Proxy target is intentionally set to be an empty object, to ensure
+  // that access to the prototype chain is not possible.
+  const proxy = new Proxy(
+    {},
+    {
+      has(_target: object, prop: string | symbol) {
+        return (
+          typeof prop === 'string' &&
+          ['request', 'on', 'removeListener'].includes(prop)
```

Comment:
How would an array of strings include a non-tring? Is the typeof check here only to prevent Typesctipt from complaining, or am I missing something?

Reply:
> Is the typeof check here only to prevent Typesctipt[sic] from complaining

Yep.

```ts
declare const prop: string | symbol;

['request', 'on', 'removeListener'].includes(prop); // TS2345: Argument of type 'string | symbol' is not assignable to parameter of type 'string'.
typeof prop === 'string' && ['request', 'on', 'removeListener'].includes(prop); // Ok.
```

Reply:
:sadpanda: